lib/global/slim.js

```diff
@@ -0,0 +1,58 @@
+var types = require("ast-types");
+var getAst = require("../get_ast");
+var concat = require("lodash/concat");
+var defaultTo = require("lodash/defaultTo");
+var compile = require("estemplate").compile;
+var optionsNormalize = require("../options_normalize");
+
+module.exports = function(load, options) {
+	var moduleId = load.uniqueId || load.name;
+	var metadata = defaultTo(load.metadata, {});
+
+	var requires = defaultTo(metadata.deps, [])
+		.map(makeNormalizer(load, options))
+		.map(toStealRequire);
+
+	var _exports = defaultTo(metadata.exports, []).map(toStealExports);
+
+	return compile(
+		`[${moduleId}, function(stealRequire, stealExports) { %= body % }]`
+	)({
+		body: concat(requires, getAst(load).body, _exports)
+	});
+};
+
+/**
+ * Turns the module id into a `stealRequire` call node
+ * @param {string} moduleId - The dependency identifier
+ * @return {Object} The AST node for the stealRequire expression
+ */
+function toStealRequire(moduleId) {
+	return compile(`stealRequire(%= moduleId %);`)({
+		moduleId: types.builders.literal(moduleId)
+	}).body[0];
+}
+
+/**
+ * Turns the property name into a `stealExports` assignment node
+ * @property {string} property - The property name to be exported
+ * @return {Object} The AST node of the `stealExports` expression
+ */
+function toStealExports(property) {
+	return compile(`stealExports["${property}"] = ${property};`)({}).body[0];
```

Comment:
I think it might be more complicated than this. Does this rely on there being a variable like `GLOBAL` ?  What if the code looks like:

```js
(function(self){

  self.GLOBAL = {};

})(this);
```

Does that work here?